__tests__/app.test.js

```diff
@@ -0,0 +1,51 @@
+const request = require("supertest");
+const db = require("../db/connection.js");
+const data = require("../db/data/test-data/index.js");
+const app = require("../app.js");
+const seed = require("../db/seeds/seed.js");
+
+afterAll(() => {
+  return db.end();
+});
+beforeEach(() => {
+  return seed(data);
+});
+
+describe("app", () => {
+  describe("/api", () => {
+    describe("invalid endpoint", () => {
+      test("status code 404", () => {
+        return request(app).get("/api/nonsense").expect(404);
+      });
+      test("404: returns message endpoint not found", () => {
+        return request(app).get("/api/nonsense").then(({body})=> {
+            expect(body.msg).toBe('Endpoint Not Found')
+        });
+      });
+    });
+    describe("GET /topics", () => {
+      test("status code 200", () => {
+        return request(app).get("/api/topics").expect(200);
```

Comment:
^^ this test could technically merge with the next test - i.e. becomes :
return request(app)
.get("/api/topics")
.expect(200)
.then(({body{) ... etc